src/Nethermind/Nethermind.Network.Discovery/Serializers/NeighborsMsgSerializer.cs

```diff
@@ -82,10 +82,10 @@ private static Node[] DeserializeNodes(RlpStream rlpStream)
         return rlpStream.DecodeArray<Node>(_decodeItem);
     }
 
-    private static int GetNodesLength(Node[] nodes, out int contentLength)
+    private static int GetNodesLength(ArraySegment<Node> nodes, out int contentLength)
     {
         contentLength = 0;
-        for (int i = 0; i < nodes.Length; i++)
+        for (int i = 0; i < nodes.Count; i++)
```

Comment:
Would be faster to [AsSpan](https://learn.microsoft.com/en-us/dotnet/api/system.memoryextensions.asspan?view=net-9.0#system-memoryextensions-asspan-1(system-arraysegment((-0))-system-index)) the `ArraySegment<Node>` and iterate the span; but if not many items probably not too worth it